rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,108 @@
+
+# Motivation
+[motivation]: #motivation
+
+The NixOS community struggles with a clearly defined way in which someone's behavior may
+be disruptive. A well-defined Code of Conduct gives an explicit
+set of expectations for all contributors, which will also make it easier to identify
+when behavior becomes disruptive.
+
+# Detailed design
+[design]: #detailed-design
+
+These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
+which provides a great compromise between explicit behaviors and subjective goals.
+Only the moderation body name has been changed.
+
+- We are committed to providing a friendly, safe and welcoming environment for
+all, regardless of level of experience, gender identity and expression,
+sexual orientation, disability, personal appearance, body size, race, ethnicity, age,
+religion, nationality, or other similar characteristics.
+- Please avoid using overtly sexual aliases or other nicknames that might
+detract from a friendly, safe and welcoming environment for all.
+- Please be kind and courteous. There’s no need to be mean or rude.
+- Respect that people have differences of opinion and that every design or
+implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
+- Please keep unstructured critique to a minimum. If you have solid ideas
+you want to experiment with, make a fork and see how it works.
+- We will exclude you from interaction if you insult, demean or harass anyone.
+That is not welcome behavior. We interpret the term “harassment” as including the definition in the
+[Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/master/citizen_code_of_conduct.md);
+if you have any lack of clarity about what might be included in that concept,
+please read their definition. In particular, we don’t tolerate behavior that excludes
+people in socially marginalized groups.
+- Private harassment is also unacceptable. No matter who you are, if you feel
+you have been or are being harassed or made uncomfortable by a community member,
+please contact one of the channel ops or any of the NixOS moderation team immediately.
+Whether you’re a regular contributor or a newcomer, we care about making this community
+a safe place for you and we’ve got your back.
+- Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+There have been a few spectacular failures of communication and good faith within the community.
+Most notable recent examples include the [block evasion thread](https://discourse.nixos.org/t/github-block-evasion-is-not-acceptable/12763),
+heated discussions in [RFC #98](https://github.com/NixOS/rfcs/pull/98) and [RFC #111](https://github.com/NixOS/rfcs/pull/111).
+Each of these incidents have caused significant discussion on IRC (when it was still official),
```

Comment:
```suggestion
Each of these incidents has caused significant discussion on IRC (when it was still official),
```

Reply:
Have is grammatically correct

Reply:
I think that's a difference between flav(o|ou)rs of English, `has` feels right to me as someone who's learned British English

Reply:
“Each” is iterating over the elements of the list no? And in every iteration the subject is singular

Reply:
Ironically, I originally wrote `has`, then my word processor recommended `have`, and I changed it because incidents is a plural word.

Reply:
Never trust the grammar check :D

Reply:
I have found online _some_ style guides that seem to be OK with «each … have», and some people here even think it flows better, so it's not so clear of a case! (But yes, I also find «has» more fitting).

Reply:
I think it makes two different past tenses, both of which may be correct in this instance.

As long as the meaning is clear, I'm not going to favor one or the other.